npeps/npep-285-combine-crds.md

```diff
@@ -0,0 +1,100 @@
+# NPEP-285: NPEP template
+
+* Issue: [#285](https://github.com/kubernetes-sigs/network-policy-api/issues/285)
+* Status: Provisional
+
+## TLDR
+
+Combine ANP and BANP into a single CRD.
+
+## Goals
+
+- Outline existing differences between ANP and BANP.
+- Merge ANP and BANP into a single CRD.
+
+## Non-Goals
+
+(What is explicitly out of scope for this proposal.)
+
+## Introduction
+
+ANP and BANP are 2 layers of policies around NetworkPolicy, evaluated as shown in the diagram below.
+
+![image](../site-src/images/ANP-api-model.png)  
+Currently, ANP and BANP are 2 separate CRDs with the following differences:
+- BANP is a singleton, which is also the reason why it doesn't have `spec.priority` field.
+- ANP supports `Pass` action, while BANP doesn't. `Pass` action could be considered a delegation to the NetworkPolicy, 
+or it could be seen as return from the current policy layer .
+- ANP supports `domainNames` matching for egress rules, while BANP doesn't.
+
+## User-Stories/Use-Cases
+
+Story 1: Multiple BANPs
+
+As a cluster admin, I want to apply different BANPs to different sets namespaces, so that I can have different default trust levels.
+For example, "system" namespaces, like kube-system, could have a less strict BANP than "user" namespaces.
+
+Story 2: Single CRD
+
+As a cluster admin, I prefer to have 1 CRD for both ANP and BANP, so that it is easier to manage, understand, and monitor.
+As an API implementation developer, I prefer to have 1 CRD for both ANP and BANP, so that I can reduce the complexity of the code.
```

Comment:
Is the code really complex? The fields in BANP are a subset of what we have for ANP

Reply:
maintaining 2 copies of something is always harder and more error-prone. How much more complex - depends on the implementation, but there is no way to implement it with 0 overhead

Reply:
I had an API maintainer user story separately, because 2 CRDs add overhead both for API maintainers and for API implementation devs. While we can ignore our "own" API maintainers problems, making implementors' lives easier sounds like a valid user story to me